src/log/iobuf.rs

```diff
@@ -13,6 +13,8 @@ struct IOBuf {
     log_offset: AtomicUsize,
 }
 
+unsafe impl Sync for IOBuf {}
```

Comment:
this basically tells the compiler "this is responsible for its own synchronization if it's shared by multiple threads"

It was necessary to add it now because in the periodic flusher, we send an `Arc` containing an `IOBufs` to another thread. We didn't need to specify this before, even though we have multithreaded tests, because `Sync` is implemented for higher-level structures that contain `IOBufs` already, and if you mark a container as `Sync`, it doesn't require that all members are recursively `Sync` as well. This is part of why it's `unsafe`, because you're basically telling the compiler "don't worry, I've got this".

This can lead to really subtle bugs, even in the standard library. This article,  https://www.ralfj.de/blog/2017/06/09/mutexguard-sync.html talks about how `MutexGuard` (the thing returned by calling `lock()` on a mutex, which implements `Deref` and `DerefMut` to pass accesses to the underlying memory contained in the `Mutex`) used to implement  `Sync`. This is bad, because if something is `Sync`, it can be sent to multiple threads simultaneously. This let you unlock a mutex, then send the result to multiple threads, even though the whole point of a mutex is to make sure multiple threads can't access something simultaneously.